discord/iterators.py

```diff
@@ -444,46 +442,28 @@ def __init__(
 
         self.entries = asyncio.Queue()
 
-        if self.reverse:
-            self._strategy = self._after_strategy
-            if self.before:
-                self._filter = lambda m: int(m["id"]) < self.before.id
-        else:
-            self._strategy = self._before_strategy
-            if self.after and self.after != OLDEST_OBJECT:
-                self._filter = lambda m: int(m["id"]) > self.after.id
+        self._strategy = self._strategy_exec
```

Comment:
If there's a single strategy, there's no need to store it as a "strategy", it can be executed directly.